### ISSUE TITLE
doesn't update automatically using crontab

### ISSUE THREAD

**Issue by triksmelb** [TomChantler/Update-Plex#3]
Unable to get this to work following these steps - any help would be appreciated.

1. Download script and run `chmod +x update-plex.sh`
2. Run `apt install jq`
3. Add token to script just to be sure
4. Add `0 2 * * * /bin/bash ./update-plex.sh` to crontab

Running the script manually works great.

**Comment by TomChantler** [comment]
It's not you, it's me. I hadn't tried running the script as a cron job. Thank you for letting me know it didn't work.

## TL;DR
Just grab the updated version of `update.plex.sh`. But please read on, as there might be more than one thing going wrong...

## Diagnosing the error
In order to diagnose the problem, I used `sudo crontab -e` (because I want the job to run as `root`) to create a crontab entry that looked like this:

```
*/5 * * * * /bin/bash /home/tom/update-plex.sh >>/tmp/log 2>>/tmp/errors
```
In other words, I copied your entry, but changed it to run every five minutes and to log the output and errors to separate files.

> NOTE: When the cron job runs as `root`, the home directory is `/root` so, in your example, where you have the path of the update script as `./update-plex.sh` you would need to have the script at the path `/root/update-plex.sh`.

The first time this ran, it failed and I looked at the files and saw the following:
### `/tmp/log`
```
Beta Channel = false
Dry Run = false
Force Download = false
Overwrite File = false
Architecture = amd64
Latest version:    1.32.2.7100-248a2daf0
Installed version: 1.32.0.6918-6f393eda1
------------------------------------------------------------
Need to upgrade...
Found latest version at https://downloads.plex.tv/plex-media-server-new/1.32.2.7100-248a2daf0/debian/plexmediaserver_1.32.2.7100-248a2daf0_amd64.deb
Downloading plexmediaserver_1.32.2.7100-248a2daf0_amd64.deb...
Installing it now...
plexmediaserver_1.32.2.7100-248a2daf0_amd64.deb installed
Done
```
From the last two lines, it looks like it worked, but it didn't (there should be more output than this). A glance in the error file showed why:
### `/tmp/errors`
```
dpkg: warning: 'ldconfig' not found in PATH or not executable
dpkg: warning: 'start-stop-daemon' not found in PATH or not executable
dpkg: error: 2 expected programs not found in PATH or not executable
Note: root's PATH should usually contain /usr/local/sbin, /usr/sbin and /sbin
```
## The Problem
It seems that the script couldn't find the necessary binaries (`ldconfig`, `start-stop-daemon`) to run `dpkg -i`. This is due to a restricted `PATH` in the context of the cron job (the environment in which cron scripts are run is typically more limited than the standard interactive shell).

## The Solution
The solution was to add the missing entries to the `PATH` variable, as it suggested in the final line of the errors file. So I modified the top of the script like this:
```
#!/bin/bash
# tomssl.com/update-plex-server-on-ubuntu-automatically
# Uncomment the next line if the script will be run as a cron job.
PATH=/usr/local/sbin:/usr/sbin:/sbin:$PATH 
plex_token=PUTYOURSHERE
beta_channel=false
# rest of script as before...
```
When I did this and tried the cron job again, nothing was logged to `tmp/errors` and `/tmp/log` had the expected output:
```
Beta Channel = false
Dry Run = false
Force Download = false
Overwrite File = false
Architecture = amd64
Latest version:    1.32.2.7100-248a2daf0
Installed version: 1.32.0.6918-6f393eda1
------------------------------------------------------------
Need to upgrade...
Found latest version at https://downloads.plex.tv/plex-media-server-new/1.32.2.7100-248a2daf0/debian/plexmediaserver_1.32.2.7100-248a2daf0_amd64.deb
File already exists, not going to download it again
Installing it now...
(Reading database ... 113367 files and directories currently installed.)
Preparing to unpack plexmediaserver_1.32.2.7100-248a2daf0_amd64.deb ...
PlexMediaServer install: Pre-installation Validation.
PlexMediaServer install: Pre-installation Validation complete.
Unpacking plexmediaserver (1.32.2.7100-248a2daf0) over (1.32.0.6918-6f393eda1) ...
Setting up plexmediaserver (1.32.2.7100-248a2daf0) ...
PlexMediaServer install: PlexMediaServer-1.32.2.7100-248a2daf0 - Installation starting.
PlexMediaServer install:
PlexMediaServer install: Now installing based on:
PlexMediaServer install:   Installation Type:   Update
PlexMediaServer install:   Process Control:     systemd
PlexMediaServer install:   Plex User:           plex
PlexMediaServer install:   Plex Group:          plex
PlexMediaServer install:   Video Group:         render
PlexMediaServer install:   Metadata Dir:        /var/lib/plexmediaserver/Library/Application Support
PlexMediaServer install:   Temp Directory:      /tmp
PlexMediaServer install:   Lang Encoding:       en_US.UTF-8
PlexMediaServer install:   Processor:           Intel(R) Pentium(R) Gold G5400T CPU @ 3.10GHz
PlexMediaServer install:   Intel i915 Hardware: Found
PlexMediaServer install:   Nvidia GPU card:     Not Found
PlexMediaServer install:
PlexMediaServer install: Completing final configuration.
PlexMediaServer install: Starting Plex Media Server.
PlexMediaServer install: PlexMediaServer-1.32.2.7100-248a2daf0 - Installation successful.  Errors: 0, Warnings: 0
Processing triggers for mime-support (3.64ubuntu1) ...
plexmediaserver_1.32.2.7100-248a2daf0_amd64.deb installed
Done
```

Thanks again for bringing this to my attention.
